src/aind_data_transfer_service/server.py

```diff
@@ -382,13 +386,67 @@ async def submit_hpc_jobs(request: Request):  # noqa: C901
     )
 
 
+async def get_job_status_list(request: Request):
+    """Get status of jobs with default pagination of limit=25 and offset=0."""
+    # TODO: Use httpx async client
+    try:
+        params = AirflowDagRunsRequestParameters.from_query_params(
+            request.query_params
+        )
+        params_dict = json.loads(params.model_dump_json())
+        response_jobs = requests.get(
+            url=os.getenv("AIND_AIRFLOW_SERVICE_JOBS_URL"),
+            auth=(
+                os.getenv("AIND_AIRFLOW_SERVICE_USER"),
+                os.getenv("AIND_AIRFLOW_SERVICE_PASSWORD"),
+            ),
+            params=params_dict,
+        )
+        status_code = response_jobs.status_code
+        if response_jobs.status_code == 200:
+            dag_runs = AirflowDagRunsResponse.model_validate_json(
+                json.dumps(response_jobs.json())
+            )
+            job_status_list = [
+                JobStatus.from_airflow_dag_run(d) for d in dag_runs.dag_runs
+            ]
+            message = "Retrieved job status list from airflow"
+            data = {
+                "params": params_dict,
+                "total_entries": dag_runs.total_entries,
+                "job_status_list": [
+                    json.loads(j.model_dump_json()) for j in job_status_list
+                ],
+            }
+        else:
+            message = "Error retrieving job status list from airflow"
+            data = {"params": params_dict, "errors": [response_jobs.json()]}
+    except ValidationError as e:
+        logging.error(e)
+        status_code = 406
+        message = "Error validating request parameters"
+        data = {"errors": json.loads(e.json())}
+    except Exception as e:
+        logging.error(e)
+        status_code = 500
+        message = "Unable to retrieve job status list from airflow"
+        data = {"errors": [f"{e.__class__.__name__}{e.args}"]}
+    return JSONResponse(
+        status_code=status_code,
+        content={
+            "message": message,
+            "data": data,
+        },
+    )
+
+
 async def index(request: Request):
     """GET|POST /: form handler"""
     return templates.TemplateResponse(
+        request=request,
```

Comment:
I'm not sure why this `request` object has been moved? It seems to break when running locally.

Reply:
I was getting DeprecationWarnings. The [docs](https://fastapi.tiangolo.com/advanced/templates/) mention that starting FastAPI 0.108.0, Starlette 0.29.0 request needs to be the first parameter rather than in the context for the jinja template.